deepchecks/nlp/text_data.py

```diff
@@ -0,0 +1,488 @@
+    @staticmethod
+    def _validate_text(raw_text: t.Sequence[str]):
+        """Validate text format."""
+        error_string = 'raw_text must be a Sequence of strings'
+        if not is_sequence_not_str(raw_text):
+            raise DeepchecksValueError(error_string)
+        if not all(isinstance(x, str) for x in raw_text):
+            raise DeepchecksValueError(error_string)
+
+    @staticmethod
+    def _validate_tokenized_text(tokenized_text: t.Sequence[t.Sequence[str]]):
+        """Validate tokenized text format."""
+        error_string = 'tokenized_text must be a Sequence of sequences of strings'
+        if not is_sequence_not_str(tokenized_text):
+            raise DeepchecksValueError(error_string)
+        if not all(is_sequence_not_str(x) for x in tokenized_text):
+            raise DeepchecksValueError(error_string)
+        if not all(isinstance(x, str) for tokens in tokenized_text for x in tokens):
+            raise DeepchecksValueError(error_string)
+
+    def _validate_and_set_label(self, label: t.Optional[TTextLabel]):
+        """Validate and process label to accepted formats."""
+        # If label is not set, create an empty label of nulls
+        if label is None:
+            self._has_label, self._label = False, [None] * len(self._text)
+            return
+
+        self._has_label = True
+        if not is_sequence_not_str(label):
+            raise DeepchecksValueError('label must be a Sequence')
+
+        if not len(label) == len(self._text):
+            raise DeepchecksValueError('label must be the same length as raw_text')
+
+        if self.task_type == TaskType.TEXT_CLASSIFICATION:
+            if all((isinstance(x, collections.abc.Sequence) and not isinstance(x, str)) for x in label):
+                self._is_multilabel = True
+                multilabel_error = 'multilabel was identified. It must be a Sequence of Sequences of 0 or 1.'
+                if not all(all(y in (0, 1) for y in x) for x in label):
+                    raise DeepchecksValueError(multilabel_error)
+                if any(len(label[0]) != len(label[i]) for i in range(len(label))):
+                    raise DeepchecksValueError('All multilabel entries must be of the same length, which is the number'
+                                               ' of classes.')
+            elif not all(isinstance(x, (str, int)) for x in label):
+                raise DeepchecksValueError('label must be a Sequence of strings or ints or a Sequence of Sequences'
+                                           'of strings or ints (for multilabel classification)')
+
+        if self.task_type == TaskType.TOKEN_CLASSIFICATION:
+            token_class_error = 'label must be a Sequence of Sequences of either strings or integers'
+            if not all(isinstance(x, collections.abc.Sequence) for x in label):
+                raise DeepchecksValueError(token_class_error)
+
+            for i in range(len(label)):  # TODO: Runs on all labels, very costly
```

Comment:
if it was numpy it would be quicker I think, can do apply() and all()

Reply:
This label is a list. Meaning, all labels are a list of lists. If we want this to be numpy, we would need to have a 2-d array. But - the labels are not of the same length, so array doesn't fit here. So it'll have to be a list of arrays and then you lose the advantage of it